pkg/tenant/db/tenants_test.go

```diff
@@ -59,58 +59,65 @@ func (s *dbTestSuite) TestCreateTenant() {
 	require.NoError(err, "could not create tenant")
 
 	// Fields should have been populated
-	require.NotEqual(uuid.Nil, tenant.ID, "expected non-zero uuid to be populated")
+	require.NotEqual("", tenant.ID, "expected non-zero ulid to be populated")
 	require.NotZero(tenant.Created, "expected tenant to have a created timestamp")
 	require.Equal(tenant.Created, tenant.Modified, "expected the same created and modified timestamp")
 }
 
 func (s *dbTestSuite) TestRetrieveTenant() {
-	// TODO: this test will change if how the key or data marshaling is changed.
 	require := s.Require()
 	ctx := context.Background()
-	tenantID := uuid.MustParse("656a9329-5412-459c-8acb-30a8914a6479")
+	tenant := &db.Tenant{
+		ID:              ulid.MustParse("01ARZ3NDEKTSV4RRFFQ69G5FAV"),
+		Name:            "example-staging",
+		EnvironmentType: "prod",
+		Created:         time.Unix(1668660681, 0).In(time.UTC),
+		Modified:        time.Unix(1668661302, 0).In(time.UTC),
+	}
 
 	s.mock.OnGet = func(ctx context.Context, in *pb.GetRequest) (*pb.GetReply, error) {
 		if len(in.Key) == 0 || in.Namespace != db.TenantNamespace {
 			return nil, status.Error(codes.FailedPrecondition, "bad Get request")
 		}
 
-		if !bytes.Equal(in.Key, tenantID[:]) {
+		if !bytes.Equal(in.Key, tenant.ID[:]) {
 			return nil, status.Error(codes.NotFound, "tenant not found")
 		}
 
-		// Load fixture from disk
-		data, err := os.ReadFile("testdata/tenant.json")
-		if err != nil {
-			return nil, status.Errorf(codes.FailedPrecondition, "could not read fixture: %s", err)
-		}
+		// Marshal the data with msgpack
+		data, err := tenant.MarshalValue()
+		require.NoError(err, "could not marshal the tenant")
+
+		// Unmarshal the data with msgpack
+		other := &db.Tenant{}
+		err = other.UnmarshalValue(data)
+		require.NoError(err, "could not unmarshal the tenant")
```

Comment:
Should we create a follow on story for msgpack fixture helpers? RE: the golden files blog post I sent you yesterday?

Reply:
Yes, I'm going to create a story about that today.